### PR TITLE
[DRAFT] Forwarder re-architected

### PR DESCRIPTION
**Pull request by yadudoc**
The forwarder and the funcx-endpoint have essentially been re-architected to expand forwarder capabilities and reduce our deployment costs. This PR adds support deploying the latest re-architected versions of the `web-service` and the `forwarder`.

* Updates to README with instructions on deployment
* forwarder deployment now uses hostPort and hostNetwork
* Both web-service and forwarder support a `use_local_image` flag in the values.yaml that uses a local container image instead of images published by github actions.

### CODE REVIEW

File: funcx/templates/web-service-deployment.yaml [hunk 1]
```diff
@@ -20,7 +20,11 @@ spec:
             value: 'postgresql://{{  .Values.postgresql.postgresqlUsername }}:{{ .Values.postgresql.postgresqlPassword }}@{{ .Release.Name }}-postgresql:5432/{{ .Values.postgresql.postgresqlDatabase }}'
       containers:
       - name: {{ .Release.Name }}-funcx-web-service
+        {{ if .Values.webService.use_local_image }}
+        image: {{ .Values.webService.local_image }}
+        {{ else }}
         image: {{ .Values.webService.image }}:{{ .Values.webService.tag }}
```
Comment (BenGalewsky) [inline funcx/templates/web-service-deployment.yaml:26]: I don't understand why you don't just override:
```
webService:
   image: my-local-image
   tag: dev
```
Then just do `docker build -t my-local-image:dev .`
As long as pull policy is `ifNotPresent` then that should do the same thing?

Comment (yadudoc) [reply funcx/templates/web-service-deployment.yaml:26]: You're right, that's a simpler solution. I'll fix that.

File: funcx/values.yaml [hunk 2]
```diff
@@ -30,10 +30,11 @@ funcx_endpoint:
 forwarder:
   enabled: true
   image: funcx/forwarder
-  tag: configurable_ic_ports
+  tag: forwarder_redesign
   pullPolicy: IfNotPresent
-  minInterchangePort: 54000
-  maxInterchangePort: 54003
+  tasksPort: 55001
```
Comment (BenGalewsky) [inline funcx/values.yaml:35]: What do these values do? It looks like the service is set up with adding 1 and 2 to commandPort value

Comment (yadudoc) [reply funcx/values.yaml:35]: Since it's just 3 ports, I thought it's simpler to specify them explicitly.

**Pull request closed without merging**